pkg/app/piped/cloudprovider/terraform/module.go

```diff
@@ -112,3 +129,45 @@ func FindArtifactVersions(tfs []File) ([]*model.ArtifactVersion, error) {
 
 	return versions, nil
 }
+
+func isLocalModule(moduleSrc string) bool {
+	return strings.HasPrefix(moduleSrc, "./") || strings.HasPrefix(moduleSrc, "../")
+}
+
+// LocalModuleSourceConverter is a converter from local module source to its URL.
+type LocalModuleSourceConverter struct {
+	// GitURL is a URL for git repository URL
+	GitURL string
+	// Branch is a git branch for the current terraform file repo.
+	Branch string
+	// RepoDir is a dir where the terraform file repo located.
+	RepoDir string
+	// AppDir is a dir where the terraform module located.
+	AppDir string
+}
+
+func NewLocalModuleSourceConverter(gitURL, branch, repoDir, AppDir string) *LocalModuleSourceConverter {
+	return &LocalModuleSourceConverter{
+		GitURL:  gitURL,
+		Branch:  branch,
+		RepoDir: repoDir,
+		AppDir:  AppDir,
```

Comment:
Maybe we don't need these fields.
The `File` struct can contain a relative path of itself from the root of the repository.